samples/zigbee/light_bulb/src/zcl_scenes.c

```diff
@@ -0,0 +1,929 @@
+		ZB_ZCL_PACKET_PUT_DATA8(payload_ptr, 1);
+
+		/* Extension set: current_level state */
+		ZB_ZCL_PACKET_PUT_DATA8(payload_ptr, p_entry->level_control.current_level);
+	}
+
+	if (p_entry->window_covering.has_current_position_lift_percentage == ZB_TRUE) {
+		LOG_INF("Append window covering fieldset");
+
+		/* Extension set: Cluster ID = Window covering */
+		ZB_ZCL_PACKET_PUT_DATA16_VAL(payload_ptr, ZB_ZCL_CLUSTER_ID_WINDOW_COVERING);
+
+		/* Extension set: Fieldset length = 1 or 2*/
+		if (p_entry->window_covering.has_current_position_tilt_percentage == ZB_TRUE) {
+			ZB_ZCL_PACKET_PUT_DATA8(payload_ptr, 2);
+		} else {
+			ZB_ZCL_PACKET_PUT_DATA8(payload_ptr, 1);
+		}
+		/* Extension set: current_position_lift_percentage state */
+		ZB_ZCL_PACKET_PUT_DATA8(payload_ptr,
+			p_entry->window_covering.current_position_lift_percentage);
+
+		if (p_entry->window_covering.has_current_position_tilt_percentage == ZB_TRUE) {
+			/* Extension set: current_position_tilt_percentage state */
+			ZB_ZCL_PACKET_PUT_DATA8(payload_ptr,
+				p_entry->window_covering.current_position_tilt_percentage);
+		}
+	}
+
+	/* Pass the updated data pointer. */
+	return payload_ptr;
+}
+
+static zb_ret_t get_on_off_value(zb_uint8_t *on_off)
+{
+	zb_zcl_attr_t *attr_desc = zb_zcl_get_attr_desc_a(
+		ZCL_SCENES_ENDPOINT,
+		ZB_ZCL_CLUSTER_ID_ON_OFF,
+		ZB_ZCL_CLUSTER_SERVER_ROLE,
+		ZB_ZCL_ATTR_ON_OFF_ON_OFF_ID);
+
+	if (attr_desc != NULL) {
+		*on_off = (zb_bool_t)ZB_ZCL_GET_ATTRIBUTE_VAL_8(attr_desc);
+		return RET_OK;
+	}
+
+	return RET_ERROR;
+}
+
+static zb_ret_t get_current_level_value(zb_uint8_t *current_level)
+{
+	zb_zcl_attr_t *attr_desc = zb_zcl_get_attr_desc_a(
+		ZCL_SCENES_ENDPOINT,
+		ZB_ZCL_CLUSTER_ID_LEVEL_CONTROL,
+		ZB_ZCL_CLUSTER_SERVER_ROLE,
+		ZB_ZCL_ATTR_LEVEL_CONTROL_CURRENT_LEVEL_ID);
+
+	if (attr_desc != NULL) {
+		*current_level = ZB_ZCL_GET_ATTRIBUTE_VAL_8(attr_desc);
+		return RET_OK;
+	}
+
+	return RET_ERROR;
+}
+
+static zb_ret_t get_current_lift_value(zb_uint8_t *percentage)
+{
+	zb_zcl_attr_t *attr_desc = zb_zcl_get_attr_desc_a(
+		ZCL_SCENES_ENDPOINT,
+		ZB_ZCL_CLUSTER_ID_WINDOW_COVERING,
+		ZB_ZCL_CLUSTER_SERVER_ROLE,
+		ZB_ZCL_ATTR_WINDOW_COVERING_CURRENT_POSITION_LIFT_PERCENTAGE_ID);
+
+	if (attr_desc != NULL) {
+		*percentage = ZB_ZCL_GET_ATTRIBUTE_VAL_8(attr_desc);
+		return RET_OK;
+	}
+
+	return RET_ERROR;
+}
+
+static zb_ret_t get_current_tilt_value(zb_uint8_t *percentage)
+{
+	zb_zcl_attr_t *attr_desc = zb_zcl_get_attr_desc_a(
+		ZCL_SCENES_ENDPOINT,
+		ZB_ZCL_CLUSTER_ID_WINDOW_COVERING,
+		ZB_ZCL_CLUSTER_SERVER_ROLE,
+		ZB_ZCL_ATTR_WINDOW_COVERING_CURRENT_POSITION_TILT_PERCENTAGE_ID);
+
+	if (attr_desc != NULL) {
+		*percentage = ZB_ZCL_GET_ATTRIBUTE_VAL_8(attr_desc);
+		return RET_OK;
+	}
+
+	return RET_ERROR;
+}
+
+static void save_state_as_scene(struct scene_table_on_off_entry *p_entry)
+{
+	if (has_cluster(ZB_ZCL_CLUSTER_ID_ON_OFF) &&
+		get_on_off_value(&p_entry->on_off.on_off) == RET_OK) {
```

Comment:
Could it be aligned to the parenthesis opening the `if` statement?

Reply:
Absolutely ;)
Fixed.